src/plugins/intel_cpu/src/node.h

```diff
@@ -774,9 +774,13 @@ class Node {
                                        NameFromType(getType()));
     }
 
+    DnnlScratchPadPtr getScratchPad() const {
+        return context->getScratchPad(curNumaNode);
+    }
+
```

Comment:
The whole idea behind the `getScratchPadMem` is to have a single entry point to access the scratchpad memory. This ensures a uniform scratchPad processing across the implementations and isolates the interaction with the scratchpad on the base level. Bypassing this single entry point increase the code complexity introducing additional dependency on the implementation level. Could you please elaborate on why it's absolutely necessary to extend the Node interface and make the Snippet executor depend on the scratchpad directly, while the memory may be request using `getScratchPadMem` call?

Reply:
Because just `Node` has the method `getScratchPadMem` while `SubgraphExecutor` responses on buffer scratchpad memory allocation. We could just pass `DnnlScratchPadPtr` to `SubgraphExecutor` which allocates memory using this structure itself.

If we shouldn't introduce `getScratchPad()` and use the existing interface, I added new method `MemoryPtr getScratchPadMem(const CpuBlockedMemoryDescPtr& desc)` as we discussed offline since Subgraph is not interested in DNNL Descriptor and can use just CPUBlocked Descriptor. No I pass lambda-function `allocator` to SubgraphExecutors instead of `DnnlScratchPadPtr`